bitcoin/contrib/test.sh

```diff
@@ -31,6 +31,12 @@ if cargo --version | grep "1\.41"; then
     cargo update -p syn --precise 1.0.107
 fi
 
+# Pin dependencies as above (required for no-std tests that use Rust 1.47 toolchain).
+if cargo --version | grep "1\.47"; then
+    cargo update -p serde --precise 1.0.156
+    cargo update -p syn --precise 1.0.107
+fi
```

Comment:
Squash with the previous change into first commit?

Reply:
I'll leave it for you to squash if wanted @apoelstra since it's your branch - I only pushed to it yesterday because I thought it was a trifling matter and I could save you some time, seems not :(